tests/integration/server/api/roles_api_test.py

```diff
@@ -403,57 +419,492 @@ def test_add_role_task():
+        )
+
+        # Create test role
+        role_resource = {
+            "name": "Men of Low Moral Fiber",
+            "owners": role_owner["next_id"],
+            "administrators": role_owner["next_id"],
+        }
+        role_response = session.post(
+            "http://rbac-server:8000/api/roles", json=role_resource
+        )
+        assert role_response.status_code == 200, (
+            "Error creating role: %s" % role_response.json()
         )
+
+        # Wait for role in db
         role_id = role_response.json()["data"]["id"]
-        proposal = add_role_member(
-            session, role_id, {"id": response2.json()["data"]["user"]["id"]}
-        )
-        conn = connect_to_db()
-        role_exists = (
-            r.db("rbac")
-            .table("roles")
-            .filter({"name": "AnaraTestRole"})
-            .coerce_to("array")
-            .run(conn)
-        )
-        assert role_exists
-
-        deletion = session.delete("http://rbac-server:8000/api/roles/" + role_id)
-        time.sleep(5)
-        proposal_result = (
-            r.db("rbac")
-            .table("proposals")
-            .filter({"proposal_id": proposal.json()["proposal_id"]})
-            .coerce_to("array")
-            .run(conn)
-        )
-        conn.close()
-        assert proposal_result[0]["status"] == "REJECTED"
-        delete_user_by_username("anara_user1")
-        delete_user_by_username("anara_user2")
+        is_role_in_db = wait_for_role_in_db(role_id)
+        assert (
+            is_role_in_db is True
+        ), "Couldn't find role in rethinkdb, maximum attempts exceeded."
+
+        # Add role member
+        role_update_payload = {
+            "id": new_member["next_id"],
+            "reason": "Integration test of membership proposal removal on role deletion.",
+            "metadata": "",
+        }
+        member_response = session.post(
+            "http://rbac-server:8000/api/roles/{}/members".format(role_id),
+            json=role_update_payload,
+        )
+        assert member_response.status_code == 200, (
+            "Error adding role member: %s" % member_response.json()
+        )
+
+        # Auth as role_owner
+        payload = {"id": role_owner["username"], "password": role_owner["password"]}
+        auth_response = session.post(
+            "http://rbac-server:8000/api/authorization/", json=payload
+        )
+        assert auth_response.status_code == 200, "Failed to authenticate as %s. %s" % (
+            role_owner["username"],
+            auth_response.json(),
+        )
+
+        # Delete test role
+        delete_role_response = session.delete(
+            "http://rbac-server:8000/api/roles/%s" % role_id
+        )
+        assert delete_role_response.status_code == 200, (
+            "Error deleting role: %s" % delete_role_response.json()
+        )
+
+        # Check for open role member proposals
+        are_proposals_rejected = wait_for_prpsl_rjctn_in_db(role_id)
```

Comment:
This function wait_for_prpsl_rjctn_in_db() is not properly checking for any open proposals for a role as it is calling fetch_relationships().